### PR TITLE
chore: improve nx config support and caching

### PR DESCRIPTION
**Pull request by JeremyJonas**
### Improve NX project and target configurations
- enable PDKProject direct nx project config
- move default nx build target to workspace
- remove legacy nx override logic
- separate out aws-arch generate task and enable caching

### Improve aws-arch generator caching
- Replace "cache dir" with "hash based" temp dir to ensure unique empty dir while somewhat minimizing requests during dev
- Add `static/**` to generate input caching

### CODE REVIEW

File: packages/nx-monorepo/src/nx-monorepo.ts [hunk 1]
```diff
@@ -458,10 +345,7 @@ export class NxMonorepoProject extends TypeScriptProject {
       .forEach((subProject: Project) => {
         // generate a package.json if not found
         const manifest: any = {};
-        (manifest.nx = this.targetOverrides[subProject.outdir]
-          ? { targets: this.targetOverrides[subProject.outdir] }
-          : undefined),
-          (manifest.name = subProject.name);
+        manifest.name = subProject.name;
```
Comment (agdimech) [inline packages/nx-monorepo/src/nx-monorepo.ts]: why is this needed?

File: packages/nx-monorepo/src/nx-types.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  export interface TargetDefaults {
+    readonly [name: string]: ProjectTarget;
+  }
+
+  /**
+   * NX workspace configurations.
+   *
+   * @see https://nx.dev/configuration/packagejson
+   */
+  export interface WorkspaceConfig {
```
Comment (agdimech) [inline packages/nx-monorepo/src/nx-types.ts:98]: Is this a breaking change given previously this was called NXConfig?

**Review by agdimech — COMMENTED** [review]
Overall it looks good and is much cleaner. Just had a few comments.

File: packages/nx-monorepo/src/nx-types.ts (hunk 2 above)
Comment (JeremyJonas) [reply packages/nx-monorepo/src/nx-types.ts:98]: Yes it is breaking, but only low-level config within nx-monorepo which should only be used by internals. As discussed I have updated the nx-monorepo README to include breaking details, along with commit message to denote such as well.

Given this is still v0, it is expected to break and we have now added sufficient messaging for potential external users.

**Comment by JeremyJonas** [comment]
@agdimech I have re-opened this stale/closed PR, which resolves build issues for https://github.com/aws/aws-prototyping-sdk/pull/266. 

I have rebased mainline and updated the commits as discussed - ready for your review.

**Comment by JeremyJonas** [comment]
@agdimech After digging a bit deeper here, the real issue is the `aws-arch` generate script is simply not deterministic due to not being able to pin pricing manifest content that is fetch from the url. All the other sources are hashed urls, so deterministic between builds.

The only way I could make this deterministic is to commit the pricing-manifest.json file to the source code.

I also added a github workflow to run weekly to auto fetch the pricing and run build (which also updates the snapshots) - currently it requires manual approval since I want to see the snapshots since there is lots of auto mapping atm.

Hopefully this works well enough.

Commit for this new code here: https://github.com/aws/aws-prototyping-sdk/pull/231/commits/26112111c16b07e61a25991cd490cf7a8fb0fdc4

**Review by agdimech — APPROVED** [review]

**Review by agdimech — APPROVED** [review]
